crates/wysiwyg/src/composer_model/quotes.rs

```diff
@@ -0,0 +1,319 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::dom::nodes::dom_node::DomNodeKind::{Generic, Quote};
+use crate::{
+    ComposerAction, ComposerModel, ComposerUpdate, DomHandle, DomNode,
+    UnicodeString,
+};
+
+impl<S> ComposerModel<S>
+where
+    S: UnicodeString,
+{
+    pub fn quote(&mut self) -> ComposerUpdate<S> {
+        if self.action_is_reversed(ComposerAction::Quote) {
+            self.remove_quote()
+        } else {
+            self.add_quote()
+        }
+    }
+
+    fn add_quote(&mut self) -> ComposerUpdate<S> {
+        let (s, e) = self.safe_selection();
+        let Some(wrap_result) = self.state.dom.find_nodes_to_wrap_in_block(s, e) else {
+            // No suitable nodes found to be wrapped inside the quote. The Dom should be empty
```

Comment:
I would change this comment to:
```
// No wrappable nodes found.
// Adding an empty Quote block with an a single ZWSP
```

Reply:
Changed it to something similar in 9f6f3bdf7afa49dfab185836bf73b47f7db0c8bb